### PR TITLE
[NUI] Make TabView APIs public

### PR DESCRIPTION
**Pull request by jaehyun0cho**
This is ACR patch to make TabView APIs public.

This pull request includes properties and methods of TabView, TabBar,
TabButton, TabButtonstyle and TabContent classes.

### API Changes ###

 - ACR: TCSACR-420

### CODE REVIEW

**Review by hyunjushin — APPROVED** [review]

**Review by Seoyeon2Kim — APPROVED** [review]
Looks good to me, but please add ACR  number. (TCSACR-xxx)

**Comment by jaehyun0cho** [comment]
ACR number is TCSACR-420

File: src/Tizen.NUI.Components/Controls/TabBar.cs [hunk 1]
```diff
@@ -30,8 +30,10 @@ public class TabButtonSelectedEventArgs : EventArgs
     {
         /// <summary>
         /// Creates a new instance of TabButtonSelectedEventArgs.
+        /// The indices of tab buttons in TabBar are basically the order of adding to TabBar by AddTabButton().
```
[the remark this answers is not in the document]
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/TabBar.cs:33]: Thank you for your comment!
I am not going to open AddTabButton() and RemoveTabButton().
So I have removed mentions in the description about AddTabButton() and RemoveTabButton().

File: src/Tizen.NUI.Components/Controls/TabBar.cs [hunk 2]
```diff
@@ -30,8 +30,10 @@ public class TabButtonSelectedEventArgs : EventArgs
     {
```
[the remark this answers is not in the document]
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/TabBar.cs:30]: Yes, you are right about the nested classes.
In this case, TabButtonSelectedEventArgs is separated class. (not to be public now)

**Comment by jaehyun0cho** [comment]
As @wiertel 's comments, the following updates have been done.

1. \<inheritdoc/\> is used on Dispose().
2. To describe index better, <see cref="TabView.AddTab"/> and <see cref="TabView.RemoveTab"/> are added to description.
3. [EditorBrowsable(EditorBrowsableState.Never)] is added to TabBar.OnRelayout() not to open it publicly.
4. To make users understand better about TabView and class related to TabView, example code is added to TabView class description.

**Comment by dongsug-song** [comment]
https://code.sec.samsung.net/jira/browse/TCSACR-420
has not been accepted yet.
and now I put the **\*\*NO MERGE\*\*** tag on this PR.
you can remove **\*\*NO MERGE\*\*** tag for yourself when this PR is ACR accepted and ready to be merged.
(**\*\*NO MERGE\*\*** tag has no restriction to submit PR or to edit this patch but just for notification and marking.
as you know, this PR can not be merged before the ACR is accepted.
thank you.

File: src/Tizen.NUI.Components/Controls/TabView.cs [hunk 3]
```diff
@@ -125,10 +179,12 @@ public void AddTab(TabButton tabButton, View view)
 
         /// <summary>
         /// Removes a tab at the specified index of TabView.
+        /// The indices of tabs(tab buttons and views) in TabView are basically the order of adding to TabView by <see cref="TabView.AddTab"/>.
+        /// So a tab(tab button and view)'s index in TabView can be changed whenever <see cref="TabView.AddTab"/> or <see cref="TabView.RemoveTab"/> is called.
```
Comment (pwasowski2) [inline src/Tizen.NUI.Components/Controls/TabView.cs:183]: Please, add a missing space and consider moving "'s" to the parentheses:

```suggestion
        /// So a tab (tab button's and view's) index in TabView can be changed whenever <see cref="TabView.AddTab"/> or <see cref="TabView.RemoveTab"/> is called.
```

Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/TabView.cs]: Thank you for your comment.
I have updated the documentation like the other documentation as follows.
```
/// So the index of a tab(tab button and view) in TabView can be changed ...
```

**Pull request merged**